Entities/Industry/Hall/Hall.as

```diff
@@ -377,33 +377,45 @@ void GetButtonsFor(CBlob@ this, CBlob@ caller)
 	if (!canSeeButtons(this, caller) || !caller.isOverlapping(this))
 		return;
 
-	if (this.getTeamNum() != 255)
+	if (!canChangeClass(this, caller))
+		return;
+	
+	if (this.getTeamNum() == 255)
+		return;
+	
+	CBitStream params;
+	params.write_u16(caller.getNetworkID());
+
+	if (!getRules().hasTag("class switching disabled"))
```

Comment:
What's preventing this (and all the other occurrences) from being moved into the `canChangeClass` function?